java/jdbc/src/main/java/io/vitess/jdbc/VitessStatement.java

```diff
@@ -707,6 +643,14 @@ protected void checkSQLNullOrEmpty(String sql) throws SQLException {
         return updateCounts;
     }
 
+    protected void checkAndBeginTransaction() throws SQLException {
+        if (!(this.vitessConnection.getAutoCommit() || this.vitessConnection.isInTransaction())) {
```

Comment:
This may be personal preference, but I find these constructs to be confusing, i.e. `!(one || two)`. Can you rewrite to `!one && !two`? That reads more naturally, in my opinion.

Reply:
Just that two negations take more CPU time than one.
I can change. Let me know.

Reply:
Ok I feel like that should be negligible relative to network time. But we can leave it.